scylla/src/statement/prepared_statement.rs

```diff
@@ -360,6 +351,93 @@ impl From<TokenCalculationError> for PartitionKeyError {
     }
 }
 
+pub(crate) type PartitionKeyValue<'ps> = (&'ps [u8], &'ps ColumnSpec);
+
+pub(crate) struct PartitionKey<'ps> {
+    partitioner_name: &'ps PartitionerName,
```

Comment:
This doesn't feel like it belongs here. Not every user of this struct will need the partitioner. I think it should just be an argument for the `calculate_token` method.